state/runtime/runtime.go

```diff
@@ -21,6 +21,7 @@ type TxContext struct {
 	ChainID      int64
 	Difficulty   types.Hash
 	Tracer       tracer.Tracer
+	NoBaseFee    bool
```

Comment:
I'd consider renaming this to `Payable` (or something like that) for 2 reasons:
- `BaseFee` in the name implies that is is related only to EIP-1559 only, which is not the case
- `NoBaseFee` then implies that logic is inverted (namely transaction is paid in case `NoBaseFee` is set to `false`, double negation basically)

Reply:
Agree. One thing is that I would have a payable state by default with an option to disable it. In this case, the flag should have `nonPayable` name.